mlflow/pyfunc/model.py

```diff
@@ -163,12 +195,22 @@ class _FunctionPythonModel(PythonModel):
     in an instance of this class.
     """
 
+    _skip_wrapping_predict = True
+
     def __init__(self, func, signature=None):
-        self.func = func
         self.signature = signature
+        # only wrap `func` if @pyfunc is not already applied
```

Comment:
Can `__init_subclass__` handle this subclass as well?

Reply:
It's a bit tricky because we need to extract type hint from self.func instead of _FunctionPythonModel.predict method, so handling the logic here is much easier

Reply:
Yea makes sense, thanks for the clarification!